### PR TITLE
Adds type checking for torque symbolizers

### LINKED ISSUE #3970: SyntaxError when unquoting -torque-aggregation-function
When removing the quotation marks from `-torque-aggregation-function` in the editor, a SyntaxError is thrown when evaluating the following function, generated by the CartoCSS parser:

```javascript
(function(data,ctx
/**/) {
var _value = null; if((8388607 & (1 << ctx.zoom))){_value = [object Object];}; return _value; 
})
```

This is related to https://github.com/CartoDB/Windshaft/issues/150 . It should either work or throw a more helpful error.

### PR DESCRIPTION
**Pull request by fdansv**
Fixes #3970, also related to https://github.com/CartoDB/Windshaft/issues/150

Prevents the editor from sending torque-specific rules with the wrong type. E.g.:

type

@javisantana

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4257/))

File: lib/assets/javascripts/cartodb/models/carto.js [hunk 1]
```diff
@@ -800,6 +800,28 @@ cdb.admin.CartoParser.prototype = {
                   index: r.index
               });
           }
+          else{
+            var type = symbolizers[r.name].type;
+            var element = r.value.value[0].value[0];
+            if(!(function(e){
```
Comment (rochoa) [inline lib/assets/javascripts/cartodb/models/carto.js]: Do you need a closure here? Can't you create a function that receives `type` and `element`?

Comment (fdansv) [inline lib/assets/javascripts/cartodb/models/carto.js]: @rochoa nah, I was being all smart about it :smile:

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4260/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4263/))

**Comment by javisantana** [comment]
what happens if I have a working style where the styles become wrong after this change? will they work?

**Comment by fdansv** [comment]
(sorry, it seems I merged just as you sent the comment, dammit!) This check is only made when styles are submitted, so it won't affect existing visualisations. Also, when the style is detected wrong, no action is applied and the last working style is the one active, as far as I know. So this shouldn't break anything already working.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4271/))

**Pull request merged**